### PR TITLE
[WIP] Cleanup

### PR DESCRIPTION
**Pull request by numberZero**
Add const/noexcept, where appropriate.
Use override.
Fix some style.

### CODE REVIEW

**Comment by paramat** [comment]
Best these cleanups are done in small chunks, to make easier review and to be mergeable before conflicts arise.

File: src/mapblock.h [hunk 1]
```diff
@@ -310,17 +318,17 @@ class MapBlock
 	//// Caller must ensure that this is not a dummy block (by calling isDummy())
 	////
 
-	inline const MapNode &getNodeUnsafe(s16 x, s16 y, s16 z)
+	inline MapNode getNodeUnsafe(s16 x, s16 y, s16 z) const noexcept
```
Comment (rubenwardy) [inline src/mapblock.h]: how large is a `MapNode`, and how often is it called to take a const& return value?

**Comment by rubenwardy** [comment]
Why is this WIP?

File: src/mapblock.h (hunk 1 above)
Comment (numberZero) [reply src/mapblock.h]: > how large is a MapNode

4 bytes.

**Comment by numberZero** [comment]
> Why is this WIP?

Because there were some errors. And, well, cleanup is always WIP by its very nature =) This PR is more of an example of what can be improved. Nevertheless, it should be mergable currently.

> I'd much prefer if C++ had something like:
> ```c++
> #pragma explicit_except
> ```

`noexcept` is way too strong to be enabled by default.

**Comment by numberZero** [comment]
@paramat So what’s the best way to post cleanups? Many tiny PRs, single branch with independent commits, plain patches?

**Comment by paramat** [comment]
Reason for close? I can understand frustration with slow dev being the reason if that is it =)

**Pull request closed without merging**